params/config.go

```diff
@@ -464,6 +487,7 @@ type Rules struct {
 	ChainID    *big.Int
 	IsIstanbul bool
 	IsLondon   bool
+	IsKIP71    bool
```

Comment:
Perhaps, we don't need to distinguish whether KIP71 hard-fork is enabled or not. How about setting `baseFee` of EVM to `0` if block header doesn't contains the `baseFee` field? Then, EVM doens't change its semantic after KIP71 hard-fork.

Reply:
I think that's good idea. but the mechanism only works outside EVM.
In the first version I developed, It seems to be necessary for EVM internal like following.

in blockchain/state_transition.go
```go
func (st *StateTransition) buyGas() error {
	mgval := new(big.Int).Mul(new(big.Int).SetUint64(st.msg.Gas()), st.gasPrice)
	mgval := new(big.Int)
	if st.evm.ChainConfig().IsKIP71ForkEnabled(st.evm.BlockNumber) {
		mgval = mgval.Mul(new(big.Int).SetUint64(st.msg.Gas()), st.evm.BaseFee)
	} else {
		mgval = mgval.Mul(new(big.Int).SetUint64(st.msg.Gas()), st.gasPrice)
	}
```
we have already included the baseFee field for ethereum equivalence. Therefore we need to distinguish hark-forked block before/after.

Reply:
@kjeom  I still think that we don't need to distinguish whether hard-forked or not. We can set the common policy for `buyGas` and `EVM` with the assumption of "baseFee will not be missing nor zero after the KIP71 hardfork". 
I think we can have a better discussion with the implementation code of KIP71 hard-fork application. Let's continue the discussion on the next PR.